### PR TITLE
[ci skip] Add note about tsrange and timezone awareness to docs

### PR DESCRIPTION
**Pull request by palkan**
According to #21116, people think that Rails knows about adapter specific TZ aware types automagically, but it's not true. I think we should add a note about PostgreSQL types.

### CODE REVIEW

File: activerecord/lib/active_record/timestamp.rb [hunk 1]
```diff
@@ -28,6 +28,10 @@ module ActiveRecord
   #
   #   ActiveRecord::Base.time_zone_aware_types = [:datetime]
   #
+  # NOTE: If you are using PostgreSQL you are likely to add <tt>tsrange</tt> and <tt>tstzrange</tt> to timezone aware types:
+  #
+  #   ActiveRecord::Base.time_zone_aware_types += [:tsrange, :tstzrange]
+  #
```
Comment (prathamesh-sonpatki) [inline activerecord/lib/active_record/timestamp.rb]: I don't think we just mention these columns in API documentation as this is not related to API.

Comment (prathamesh-sonpatki) [inline activerecord/lib/active_record/timestamp.rb]: May be we can add a generic note in configurations guide not just talking about Postgres specific types.

Comment (palkan) [inline activerecord/lib/active_record/timestamp.rb]: Ok, we can re-write this note as an example, but unless PostgreSQL adapter is a part of ActiveRecord, I think, we should provide specific information, not generic. This can help us to avoid bug reports/ issues like "Help! tsrange not working" etc.

Comment (prathamesh-sonpatki) [inline activerecord/lib/active_record/timestamp.rb]: Sure, we can replace this with, 
```
You can also add database specific timezone aware types. For eg. for PostgreSQL:

ActiveRecord::Base.time_zone_aware_types += [:tsrange, :tstzrange]

```

**Comment by maclover7** [comment]
:+1: Thank you for your contribution!

**Comment by maclover7** [comment]
:heart:
